xtext.cpp

```diff
@@ -1176,7 +1179,26 @@ class XText {
             m_charinfo[idx_to_substitute_with_ellipsis].flags &= ~CHAR_CAN_EXTEND_WIDTH;
             m_charinfo[idx_to_substitute_with_ellipsis].flags &= ~CHAR_CAN_EXTEND_WIDTH_FALLBACK;
             m_charinfo[idx_to_substitute_with_ellipsis].flags &= ~CHAR_IS_CLUSTER_TAIL;
-                // Looks like we can keep all other flags as is
+            m_charinfo[idx_to_substitute_with_ellipsis].flags &= ~CHAR_SCRIPT_CHANGE; // ellipsis is script neutral
+                                                // Looks like we can keep all other flags as is
+            if ( m_has_bidi ) {
+                // Be sure UAX#9 rules I1, I2, L1 to L4 don't move the ellipsis away
+                // from its neighbours
+                orig_idx_bidi_ctype = m_bidi_ctypes[idx_to_substitute_with_ellipsis];
+                orig_idx_bidi_level = m_bidi_levels[idx_to_substitute_with_ellipsis];
+                // If we're substituting start or end (which is what we do in frontend),
+                // set the bidi level of our ellipsis to be start+1 or end-1, so
+                // it's not moved away from next or previous char.
+                if ( idx_to_substitute_with_ellipsis == end-1 && end-1 >= start)
+                    m_bidi_levels[idx_to_substitute_with_ellipsis] = m_bidi_levels[end-1];
```

Comment:
I tend to prefer explicit `{}` so that you can't accidentally introduce mistakes and can easily add something without having to worry about adding that stuff.

Reply:
Yes, me too. But sometimes, in some context, for small things, I prefer without, to make these small things span as few lines as possible :)
Like when I could have used `value = condition ? value_yes : value no` but the items are too long to fit on a line, I'm fine with a if/else without `{ }` - rather than making the `? : ` multi-lines, which is less readable.

Reply:
Sure, I like ternary operators too. :-)